### PR TITLE
PHPC-415: Try authenticating once and include failure message

### PR DESCRIPTION
**Pull request by bjori**

### CODE REVIEW

**Comment by bjori** [comment]
Oh. This passes on travis because we skip x509 auth there.

This however does require mongoc bump after https://jira.mongodb.org/browse/CDRIVER-836

File: tests/connect/standalone-x509-error-0001.phpt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+$m2 = connect($dsn, $opts);
+
+echo "Both should have failed with auth failure - without reusing previous stream\n";
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECTF--
+MongoDB\Driver\Exception\AuthenticationException: auth failed
+MongoDB\Driver\Exception\AuthenticationException: auth failed
+Both should have failed with auth failure - without reusing previous stream
```
Comment (jmikola) [inline tests/connect/standalone-x509-error-0001.phpt]: How would this test have differed without the corresponding change to `phongo_stream_destroy()` and `phongo_stream_close()`?

Comment (bjori) [inline tests/connect/standalone-x509-error-0001.phpt]: Probably would have segfault while accessing properties of the stream after already destroying/freeing it.

**Comment by jmikola** [comment]
With [CDRIVER-836](https://jira.mongodb.org/browse/CDRIVER-836) merged last night, are you good to bump the submodule and merge this PR?

**Comment by bjori** [comment]
https://jira.mongodb.org/browse/CDRIVER-850 was also a blocker.
I think bumping the submodule now should work fine (along with with merging the prs), although I have not confirmed it.

**Comment by jmikola** [comment]
Commits are cherry-picked into #119.

**Pull request closed without merging**